dagobah/core/job.py

```diff
@@ -439,8 +439,11 @@ def _serialize(self, include_run_logs=False, strict_json=False):
             result = json.loads(json.dumps(result, cls=StrictJSONEncoder))
         return result
 
-    def _implements_function(obj, functions):
+    def _implements_function(self, obj, functions):
         """ Checks for the existence of a method or list of methods """
+        if isinstance(functions, str):
+            functions = [functions]
```

Comment:
The function is definitely simpler (and harder for people in the future to screw up) if it's always operating on one type. Since we don't need the list functionality right now, I think we should just make it always accept a single string. This is also pretty easy to compose together later on if something needs to check multiple:

```python
all(map(partial(self._implements_function, obj), ['function1', 'function2']))
```

Could either do it that way, which I realize looks terrible to anyone who's not a LISP nerd, or write another function at the same level of `_implements_function` which explicitly operates on lists and calls the single function version to do its work.